src/main/java/Katleen.java

```diff
@@ -0,0 +1,207 @@
+        String text = "";
+        while (!text.equals("bye")) {
+            text = in.nextLine();
+            System.out.println(LINE);
+            parseString(text);
+            System.out.println(LINE);
+        }
+    }
+
+    private static void parseString(String input) {
+        String[] splitInput = input.split(" ");
+        String cmdWord = splitInput[0];
+        int taskCount = tasks.size();
+        try {
+            switch (cmdWord) {
+            case "list":
+                if (tasks.isEmpty()) {
+                    System.out.println("You have no tasks!");
+                    break;
+                }
+                System.out.println("Here are the tasks in your list:");
+                int count = 1;
+                for (Task t : tasks) {
+                    System.out.print(count + ". ");
+                    t.printTask();
+                    count++;
+                }
+                break;
+            case "bye":
+                for (Task t : tasks) {
+                    FileWriter fw = new FileWriter(filename);
+                    fw.write(t.toString());
+                }
+                System.out.println(LINE);
+                System.out.println("Bye, have a nice day!");
+                System.out.println(LINE);
+                break;
+            case "mark":
+            case "unmark":
+            case "delete":
+                String index = splitInput[1];
+                int i = Integer.parseInt(index) - 1;
+                if (cmdWord.equals("delete")) {
+                    Task temp = tasks.get(i);
+                    tasks.remove(i);
+                    System.out.println("Task deleted: " + temp);
+                    break;
+                }
+                if (cmdWord.equals("mark")) {
+                    tasks.get(i).setDone(true);
+                    break;
+                }
+                tasks.get(i).setDone(false);
+                break;
+            case "todo":
+            case "deadline":
+            case "event":
+                Task task = getTask(input, cmdWord, splitInput, true);
+                if (task == null) {
+                    throw new UnrecognizedCommandException();
+                }
+                tasks.add(task);
+                tasks.set(taskCount, task);
+                tasks.get(taskCount).printTask();
+                break;
+            default:
+                throw new UnrecognizedCommandException();
+            }
+        } catch (IncompleteTaskException e) {
+            System.out.println("Excuse me, you're missing the deadline or duration.");
+        } catch (UnrecognizedCommandException e) {
+            System.out.println("Invalid command, please try again");
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static Task getTask(String input, String cmdWord, String[] splitInput, boolean saveToFile) throws IncompleteTaskException {
```

Comment:
very long method here as well, do check through your code for more long methods to shorten